### PR TITLE
add placeholder step to beat new challenges

### PR DESCRIPTION
**Pull request by merwanehamadi**
### Background

- the first step currently runs all the regression tests. It doesn't try to beat the unbeaten challenges.
- I have another PR coming where I will run the unbeaten challenges, to try to see if the prompt was improved.
- to try to run the unbeaten challenges we need to know the prompts have changed, otherwise we're going to run the pipeline for veeeery long for no reason and waste people's time when they add just a one liner in a contribution.md or a remove an unused import statement.
### Changes

- Prepare the ground for a step called "Beat new challenges" that will attempt to beat all the challenges not beaten.
- As you can see the step will only be triggered if the user changed the cassettes either prior to creating the PR OR while running the tests. (for now, until we find a better indicator)
- this step doesn't do anything for now.
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run `black .` and `isort .` against my code to ensure it passes our linter.

### CODE REVIEW

**Comment by dschonholtz** [comment]
Looks good. I am continually concerned by running arbitrary folks GitHub actions in pipelines. But the logic here seems sound. I might try to do some looking into this to see how it works

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -115,13 +130,6 @@ jobs:
...
-
-      - name: Check PR
-        if:  ${{ env.TIMESTAMP_COMMIT != null }}
-        run: |
-          echo "Pull Request Number - ${{ steps.create_pr.outputs.pull-request-number }}"
-          echo "Pull Request URL - ${{ steps.create_pr.outputs.pull-request-url }}"
-
       - name: Comment PR URL in the current PR
         if:  ${{ env.TIMESTAMP_COMMIT != null }}
         uses: thollander/actions-comment-pull-request@v2
```
Comment (dschonholtz) [inline .github/workflows/ci.yml:133]: This is the piece that weirds me out, but I understand that isn’t what this pr is about

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -115,13 +130,6 @@ jobs:
           title: "Update cassette-diff-PR${{ github.event.pull_request.number }}-${{ env.TIMESTAMP_COMMIT }}"
           body: "This PR updates the cassettes. Please merge it."
 
-
-      - name: Check PR
```
Comment (dschonholtz) [inline .github/workflows/ci.yml:119]: I’m confused as to why this is getting taken out here. 

This is to create the pr with new cassettes? Or is it something else? I understand it should be removed

**Review by dschonholtz — DISMISSED** [review]
Left a few comments I would love clarification on but this looks good

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dschonholtz) [reply .github/workflows/ci.yml:119]: I just don’t understand what it originally did and what it’s original intention was

Comment (waynehamadi) [reply .github/workflows/ci.yml:119]: just display whether the cassette  branch was created, but then I realize it's not that useful.

**Comment by merwanehamadi** [comment]
@dschonholtz I am going to put back to draft, I found a way to test the repo directly without making merges to master.

**Pull request closed without merging**